### ISSUE TITLE
Colab Pro+ runtime advice

### ISSUE THREAD

**Issue by athavalechem** [martinpacesa/BindCraft#83]
Hi Bindcraft team, congratulations for this wonderful pipeline and for opening it up to the community.

I decided to use this on a ColabPro+ account with a ~100aa target sequence. The following observations may be of interest to others who want to use Colab for executing Bindcraft. In the 24 hr runtime on a A100 GPU, about 270 compute units were utilized. 79 trajectories were run and 43 designs passed all filters. I assume this (the timeframe) is along expected lines? 

My issue now is that since ColabPro+ times out after the 24h runtime, the code has now stopped. Is there any way I can resume from the current position and if not, what would be the best way to proceed? I think everything was running fine and I do not want to lose out on the progress made so far.

**Comment by martinpacesa** [comment]
Hello, yes you can just continue generating into the same folder, it will keep adding designs

**Comment by finexzhi** [comment]
I think the question is that the 24h runtime limit will interrupt the calculation in the middle. You can stop the runtime after the end of one run and restart a new 24h runtime (reset the clock). In this way, you will not be worried about the interruption of a run.

**Comment by martinpacesa** [comment]
I don't think it matters. Yes you will lose one trajectory by restarting but it's no big deal.

**Comment by athavalechem** [comment]
Update: A new runtime can be started with all the folder paths identical and new binder designs will simply be added without overwriting anything. No problems there. The trajectory which was interrupted is lost but that is totally fine. I got the 100 designs now.
I am curious about your thoughts on the following things:
This target (~100aa, mostly helices) took ~48 hrs on ColabPro+ A100 GPU with ~560 compute units. 160 trajectories were tested to generate 101 designs passing all filters (4 stage, all settings were default values).
Questions:
1) Is this the typical time or is it too slow/fast? Are things faster on a local installation?
2) Does the ratio of trajectories/designs give a sense of how easy the target was and how good the designs may be?
3) Roughly, how are these times expected to change for targets with comparable 'complexity' but that are larger - let's say 300 aa, 800aa, 1400aa?

**Comment by martinpacesa** [comment]
1. So the Colab version is certainly much slower than a local installation. It also depends on which GPU you are using. A100 is I think the fastest you can get on colab and is certainly one of the better ones you can have locally as well. H100s are about 5x faster but much more expensive to buy.
2. This is hard to say. We had hard targets and easy targets and generally whatever passed filters also worked in the wetlab, not necessarily better or worse. But in terms of in silico success, your target is definitely amongst the best, we rarely have to sample so little (least we did was 200, most we had to do was 3500).
3. anywhere from 5 minutes to 3 hours per trajectory. Maximum we could test was 500 amino acids at that time. We can now do bigger things with larger GPU memory but the run time in some cases can be half a day per trajectory.

**Comment by athavalechem** [comment]
Thanks for all the insights - this is super useful!
